moveit_core/utils/include/moveit/utils/robot_model_builder.h

```diff
@@ -0,0 +1,178 @@
+/*********************************************************************
```

Comment:
is this intended to be used anywhere that is not test-specific? if not, rename file to something like ``robot_model_test_utils.h``

if you intend to use it outside of test, i'm concerned it has duplicate functionality with 
https://github.com/ros-planning/moveit/tree/melodic-devel/moveit_ros/planning/rdf_loader
https://github.com/ros-planning/moveit/tree/melodic-devel/moveit_ros/planning/robot_model_loader

Reply:
I didn't do a deep dive, but do you think can we get rid of the rdf_loader after this is added? Or are they very different things? I haven't looked in a few years

Reply:
I wouldn't say they're very different, rdf_loader seems to be a bit more general, looks like it handles Xacro files as well. It wouldn't be hard to move rdf_loader into moveit_core/utils and merge with `loadTestingRobotModel()` et al. though.  Should that be another PR, or should it go it here?

Reply:
Looking at it further, the rdf_loader has more ROS dependencies than allowed in moveit_core, e.g.

```
#include <ros/ros.h>
#include <ros/package.h>
```

For example it uses the parameter server. While I am conflicted about moveit_core's level of ROS-agnotistism (not very, but still a little) I think that's a discussion for another issue/PR. I'd say let's move with this PR as-is